### ISSUE TITLE
Exports not found when using absolute import paths based on baseUrl from tsconfig.json or jsconfig.json

### ISSUE THREAD

**Issue by iulian-radu-at** [import-js/eslint-plugin-import#2236]
Hi,

All my exports from .tsx files are reported as not being used when I use absolute paths based on baseUrl set in tsconfig.json

How to reproduce this problem:
1. npx create-react-app my-app --template typescript
2. cd my-app && npm install eslint-plugin-import
3. define a baseUrl of "." in tsconfig.json (in "compilerOptions")
4. create a 2 folders in "src/" (name them "a" and "b")
5. create a file in each of them ("a/out.tsx" and "b/in.tsx")
6. export a function from "a/out.tsx" (fn)
7. import fn in "b/in.tsx" using an absolute path ("import {fn} from 'src/a/out';")
8. check in "a/out.tsx" if eslint-plugin-import reports "fn" as not being used
9. you can add a script in package.json for running the linter ("lint": "eslint . --ext .ts,.tsx")

eslint is configured in package.json like this:
```json
{
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest",
      "plugin:import/typescript"
    ]
    "settings": {
      "import/resolver": {
        "node": {
          "extensions": [
            ".ts",
            ".tsx"
          ],
          "paths": [
            "."
          ]
        }
      }
    },
    "rules": {
      "import/no-unused-modules": [
        "warn",
        {
          "missingExports": true,
          "unusedExports": true,
          "src": [
            "."
          ],
          "ignoreExports": [
            "src/**/*.d.ts"
          ]
        }
      ]
    }
  },
}
```

Maybe I do something wrong in the way I configure eslint or eslint/eslint-plugin-import has a problem with absolute paths based on baseUrl.
Any help is highly appreciated. Thank you.

Cheers,
Iulian

**Comment by ljharb** [comment]
I'm not familiar with `baseUrl`, but if TS supports it we should as well.

This might need a fix here, or it might need one in `tsconfig-paths`.

**Comment by iulian-radu-at** [comment]
I think that first step should be that someone writes a test which reproduces this problem.

**Comment by jescowuester** [comment]
How does this work with `jsconfig.json`? (When not using typescript)

**Comment by ljharb** [comment]
That's not a standard thing (just a niche convention), so I would expect that it doesn't work with it.

If someone built a custom import plugin resolver for jsconfig.json, then that should work I suppose, but unless babel and bundlers read that file, i'm not sure it'd be a good idea to rely on resolution from it.

**Comment by jescowuester** [comment]
Good to know. The current workaround I'm using is just adding this to my eslint config:
```
  "settings": {
    "import/resolver": {
      "node": {
        "paths": ["src"]
      }
    }
  }
  ```

**Comment by JounQin** [comment]
You should use https://github.com/import-js/eslint-import-resolver-typescript instead
